tests/unit/test_semver.py

```diff
@@ -0,0 +1,210 @@
+from repo2docker.buildpacks.julia import semver
+
+
+def test_str_to_version():
+    assert semver.str_to_version("1.5.2") == (1, 5, 2)
+    assert semver.str_to_version("1") == (1,)
+
+
+def test_major_minor_patch():
+    V = (1, 2, 3)
+    assert (semver.major(V), semver.minor(V), semver.patch(V)) == (1, 2, 3)
+    assert semver.major((1,)) == 1
+    assert semver.minor((1,)) == 0
+    assert semver.patch((1,)) == 0
+
+
+def test_simple_matches():
+    assert repr(semver.create_semver_matcher("1.2.5")) == "[1.2.5-2)"
```

Comment:
Is there a minimal change we could make to remove the need for using `repr` so much in the tests?

It somehow doesn't feel right. The best idea I have is to teach the matcher classes how to compare themselves to a version string or version tuple in `__eq__`. But I also feel like we are drifting off into the finer points of Python and it might not be worth the time (-> let's make a low priority/getting started issue out of this).